src/containers/TransactionPage/TransactionPage.duck.js

```diff
@@ -479,11 +472,16 @@ export const sendReview = (role, tx, reviewRating, reviewContent) => (dispatch,
 
 // loadData is a collection of async calls that need to be made
 // before page has all the info it needs to render itself
-export const loadData = params => dispatch => {
+export const loadData = params => (dispatch, getState) => {
   const txId = new UUID(params.id);
-
-  // Clear the send error since the message form is emptied as well.
-  dispatch(resetState());
+  const state = getState().TransactionPage;
+  const txRef = state.transactionRef;
+
+  // In case a transaction reference is found from a previous
+  // data load -> clear the state. Otherwise keep the non-null
+  // values which may have been set from a previous page.
+  const initialValues = txRef ? {} : pickBy(state, v => !!v);
```

Comment:
Arrays are a problem here.
`!![]` => true

`v => (Array.isArray(v) && !!v.length) || !!v`

Reply:
@Gnito I thought about that and reasoned that if the state has en empty array it can be passed as an initial value as our initial value for `messages`, which seems to be the only array, is `[]`.

However, what you proposed would be more explicit. Also empty objects are truthy too and for example there's the `transactionRef` which is an object and should be overridden with the value from `initialState` in case it is somehow has been set to `{}`.

Reply:
Also, passing `[]` to

`(Array.isArray(v) && !!v.length) || !!v`

resolves to `true`